### PR TITLE
First iteration of the ControlPanel on the left of the Background [DISCL-282]

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

File: apps/DisplayCluster/MasterWindow.h [hunk 1]
```diff
@@ -83,9 +83,18 @@ class MasterWindow : public QMainWindow
     /** Emitted when users want to hide the dock. */
     void hideDock();
 
+    /** Emitted when users want to open a session. */
```
Comment (tribal-tec) [inline apps/DisplayCluster/MasterWindow.h]: ...open a content.

File: apps/DisplayCluster/resources/MasterDisplayGroup.qml [hunk 2]
```diff
@@ -14,4 +16,19 @@ DisplayGroup {
...
     }
+
+    controlPanel.buttonDelegate: Component {
+        ControlPanelDelegate {
+            id: touchControlPanel
+            TouchMouseArea {
+                anchors.fill: parent
+                onTap: {
+                    var action = touchControlPanel.ListView.view.model.get(index).action
+                    var absPos = mapToItem(dispGroup, controlPanel.width + 180, 0)
```
Comment (tribal-tec) [inline apps/DisplayCluster/resources/MasterDisplayGroup.qml]: magic 180 to style.js?

File: dc/core/CMakeLists.txt [hunk 3]
```diff
@@ -77,6 +78,7 @@ list(APPEND DCCORE_PUBLIC_HEADERS
   PixelStreamContent.h
   PixelStreamSegmentRenderer.h
   QmlWindowRenderer.h
+  qmlUtils.h
```
Comment (tribal-tec) [inline dc/core/CMakeLists.txt]: QmlUtils.h?

File: dc/core/ContentWindow.h [hunk 4]
```diff
@@ -110,12 +111,20 @@ class ContentWindow : public Coordinates
...
+    {
+        DEFAULT,    // A regular window
+        PANEL       // A panel window - always interative, cannot be focused
+    };
+    Q_ENUMS( WindowType )
+
     /**
      * Create a new window.
      * @param content The Content to be displayed.
      * @note Rank0 only.
```
Comment (tribal-tec) [inline dc/core/ContentWindow.h]: param type doxygen

**Comment by tribal-tec** [comment]
Can you post a screenshot how the control panel looks like?

**Comment by tribal-tec** [comment]
Not much (unit)tests though for the new code.

**Comment by rdumusc** [comment]
You're right, test coverage is not so good... :-( Half of the code is UI stuff which I don't know how to test, but I will add more for the cpp parts at least.

File: dc/core/CMakeLists.txt (hunk 3 above)
Comment (rdumusc) [inline dc/core/CMakeLists.txt]: Since it's not a class but a set of functions, I used lowercase (like types.h)

**Comment by rdumusc** [comment]
control panel

**Comment by rdumusc** [comment]
The design will change a lot after yesterday's discussion with Nicolas. The panel will be much smaller on the left,  with just 2-3 icon max to switch presentation mode and open a launcher panel (new qml dock + app launcher + sessions panel combined).

**Comment by rdumusc** [comment]
other comments addressed, I will upload a new patch with more unit tests later

File: dc/core/CMakeLists.txt (hunk 3 above)
Comment (tribal-tec) [inline dc/core/CMakeLists.txt]: I never understood that reasoning, but I give up then

**Comment by rdumusc** [comment]
Updated with better unit test coverage

**Comment by tribal-tec** [comment]
+1

**Comment by rdumusc** [comment]
retest this please

**Comment by rdumusc** [comment]
retest this please

**Comment by rdumusc** [comment]
Finally..!

**Comment by tribal-tec** [comment]
Yeah! +2

**Pull request merged**